src/ripple/basics/impl/contract.cpp

```diff
@@ -39,6 +41,18 @@ accessViolation() noexcept
 
 } // detail
 
+void
+LogThrow (std::string title)
+{
+    if (auto j = debugJournal().debug())
+    {
+        j << title;
+        for (auto const& frame : beast::getStackBacktrace ())
+            j << "    " << frame;
+        j << "----";
+    }
+}
```

Comment:
Yes, here is the libc++ `execute_native_thread_routine` equivalent:
https://github.com/llvm-mirror/libcxx/blob/master/include/thread#L343-L379
The crucial difference is that the libc++ version has no top level `try/catch`.  Unhandled exceptions are literally unhandled and another library (the unwind library) is responsible for calling `terminate` in this circumstance.  When it does, it does so without having unwound the stack, and thus one gets a stack trace from the point of `throw`.

I can think of no technical reason why gcc could not do this as well, because both platforms follow the same Itanium ABI specification for exception throwing.